.github/workflows/ci.yml

```diff
@@ -388,6 +389,74 @@ jobs:
       run: cmake --build build --target test_cmake_build
 
 
+  # Testing on ICC using the oneAPI apt repo
+  icc:
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+
+    name: "🐍 3 • ICC latest • x64"
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Add apt repo
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y wget build-essential pkg-config cmake ca-certificates gnupg
+        wget https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
+        sudo apt-key add GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
+        echo "deb https://apt.repos.intel.com/oneapi all main" | sudo tee /etc/apt/sources.list.d/oneAPI.list
+
+    - name: Add ICC & Python 3
+      run: sudo apt-get update; sudo apt-get install -y intel-oneapi-compiler-dpcpp-cpp-and-cpp-classic cmake python3-dev python3-numpy python3-pytest python3-pip
+
+    - name: Update pip & pytest
+      shell: bash
+      run: |
+        set +e; source /opt/intel/oneapi/setvars.sh; set -e
+        python3 -m pip install --upgrade pip
+        python3 -m pip install --upgrade pytest==5.4.3
```

Comment:
This looks odd. Why is pytest 6 not causing problems on other CI builds? Are we relying on a breaking change of pytest 6?

Reply:
First bullet point [here](https://docs.pytest.org/en/stable/changelog.html#features) seems relevant.

Reply:
We pin pytest; most CI jobs get the same one or an older one. So it's quite possible that we don't check on PyTest 6. That first bullet point looks relevant indeed, maybe we should try turning off that feature via the method mentioned in the log. Conditionally for Pytest 6, I expect 🤦.

Reply:
So, this should probably `pip install test/requirements.txt`?

Reply:
I'd suggest going for this. I just confirmed pytest 6 also fails for all other tests, when updating `tests/requirements.txt`: #2741

Reply:
Looks good, I will just rebase this PR and add the `pip install -r tests/requirements.txt` line